src/roswire/ros2/reader.py

```diff
@@ -0,0 +1,141 @@
+import json
+import os
+import shlex
+import typing
+from typing import Any, Dict, Optional, Sequence, Tuple
+
+import attr
+import pkg_resources
+from loguru import logger
+
+from ..proxy.roslaunch.config import ExecutableType, LaunchConfig, NodeConfig
+from ..proxy.roslaunch.context import LaunchContext
+from ..proxy.roslaunch.reader import LaunchFileReader
+
+if typing.TYPE_CHECKING:
+    from ..app import AppInstance
+
+
+@attr.s(eq=False)
+class ROS2LaunchFileReader(LaunchFileReader):
+
+    app_instance: 'AppInstance'
+
+    def locate_node_binary(self, package: str, node_type: str) -> str:
+        pass
+
+    def read(self,
+             filename: str,
+             argv: Optional[Sequence[str]] = None
+             ) -> LaunchConfig:
+        """Produces a summary of the effects of a launch file.
+
+           Parameters
+           ----------
+           filename: str
+               The name of the launch file, or an absolute path to the launch
+               file inside the container.
+           package: str, optional
+               The name of the package to which the launch file belongs.
+           argv: Sequence[str], optional
+               An optional sequence of command-line arguments that should be
+               supplied to :code:`roslaunch`.
+
+           Raises
+           ------
+           PackageNotFound
+               If the given package could not be found.
+           LaunchFileNotFound
+               If the given launch file could not be found in the package.
+           """
+
+        # Copy resources/launch_extractor._py as a python
+        # file into the container
+        logger.debug("Copying launch extraction script")
+        files = self.app_instance.files
+        host_script = pkg_resources. \
+            resource_filename('roswire',
+                              'resources/launch_extractor._py')
+        logger.debug(f'Resource is {host_script}')
```

Comment:
Unclear what information is being provided here. You could (a) add a bit more detail, or (b) drop it entirely.

Reply:
Removed